packages/node/src/BacktraceClient.ts

```diff
@@ -25,9 +27,48 @@ export class BacktraceClient extends BacktraceCoreClient {
             undefined,
             new VariableDebugIdMapProvider(global as DebugIdContainer),
         );
+        this.captureUnhandledErrors();
     }
 
     public static builder(options: BacktraceConfiguration): BacktraceClientBuilder {
         return new BacktraceClientBuilder(options);
     }
+
+    private captureUnhandledErrors() {
+        const unhandledRejectionMode = NodeOptionReader.read('unhandled-rejections');
+        const shouldContinueExecution = unhandledRejectionMode === 'warn' || unhandledRejectionMode === 'none';
+
+        process.prependListener(
+            'uncaughtExceptionMonitor',
+            async (error: Error, origin?: 'uncaughtException' | 'unhandledRejection') => {
+                // all rejected promises will be captured via unhandledRejection handler
+                if (origin === 'unhandledRejection') {
+                    return;
+                }
+                await this.send(new BacktraceReport(error, { 'error.type': 'Unhandled exception' }));
+            },
+        );
+
+        process.prependListener('unhandledRejection', async (reason) => {
+            const error =
+                reason instanceof Error
+                    ? reason
+                    : typeof reason === 'string'
+                    ? new Error(reason)
+                    : new Error('Unhandled rejection');
+            await this.send(
+                new BacktraceReport(error, {
+                    'error.type': 'Unhandled exception',
+                }),
+            );
```

Comment:
ANY exception here will cause a loop - let's say Node randomly throws one. Can we just wrap it here with try/catch? There's no benefit of letting our errors be catched again.

Reply:
One more important rule. Error monitoring client CANNOT throw an error. It can cause an overall infinity loop of errors. We don't want to throw any error from the error flow. We can in other flows like from database operations that the user will execute.